pkg/parse/parser.go

```diff
@@ -64,10 +64,9 @@ func (f AnyFormat) String() string {
 //
 // To get some insights about the parsing process, you can pass an `AnyDetails` pointer as an
 // argument.
-func UnmarshalAny(buf []byte, anyDetails ...*AnyDetails) (*ubp.Blueprint, error) {
-	details := &AnyDetails{}
-	if len(anyDetails) > 0 {
-		details = anyDetails[0]
+func UnmarshalAny(buf []byte, details *AnyDetails) (*ubp.Blueprint, error) {
+	if details == nil {
+		details = &AnyDetails{}
```

Comment:
Yes totally, my thinking was keeping it "compatible" with other unmarshal interfaces but the behavior of this one is so special anyway that being explicit is good.